src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -35,6 +40,64 @@ public IndexModel createIndex(CreateIndexRequest createIndexRequest) throws Pine
         return indexModel;
     }
 
+    public <E extends Enum<E>> boolean validateEnums(String stringToValidate, List<E> validEnums) {
+        return validEnums.stream()
+                .anyMatch(validEnum -> Objects.equals(stringToValidate, validEnum.toString()));
+    }
+
+    public IndexModel createServerlessIndex(String indexName, String metric, int dimension, String cloud,
+                                            String region) {
```

Comment:
I like these as simple string types for simplicity. Can we add javadoc to this method? I bet copilot will write most of it for you.

Reply:
I might leave this to @rohanshah18 since I think he's been working on that for this SDK in general, but I'll confirm in Slack :)

Reply:
Good callout though, yeah we need to add javadoc comments all over.